cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsPipelineSpec.java

```diff
@@ -38,14 +38,16 @@ public class DataStreamsPipelineSpec extends PipelineSpec {
   private final boolean isUnitTest;
   private final String checkpointDirectory;
   private final String pipelineId;
+  private final boolean nativeStateTrackingEnabled;
 
   private DataStreamsPipelineSpec(Set<StageSpec> stages, Set<Connection> connections,
                                   Resources resources, Resources driverResources, Resources clientResources,
                                   boolean stageLoggingEnabled, boolean processTimingEnabled, long batchIntervalMillis,
                                   String extraJavaOpts, int numOfRecordsPreview,
                                   boolean stopGracefully, Map<String, String> properties,
                                   boolean checkpointsDisabled, boolean isUnitTest, String checkpointDirectory,
-                                  String pipelineId, Set<String> connectionsUsed, Engine engine) {
+                                  String pipelineId, Set<String> connectionsUsed, Engine engine,
+                                  boolean nativeStateTrackingEnabled) {
```

Comment:
I don't particularily love that `checkpointDisabled = true` means "checkpoint = no" while `nativeStateTrackingEnabled = true` means "native state tracking = yes". Does it make sense to make these both either "enabled" or "disabled?"

Reply:
Considering the feature flag convention, I think it is good to keep the suffix as 'enabled'. Renaming checkpoint variable is possible, checking if that change is risky to have its own PR.

Reply:
Reverted this commit since the state is being moved to a different class. The new approach will not have this issue.